include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -351,9 +346,12 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 
 						// Now add the mainFileContents and produce both lookup and early equality rejection hashes
 						hashable.insert(hashable.end(), mainFileContents.begin(), mainFileContents.end());
-						hash = nbl::core::XXHash_256(hashable.data(), hashable.size());
-						lookupHash = hash[0];
-						for (auto i = 1u; i < 4; i++) {
+
+						core::blake3_hasher hasher;
+						hasher.update(hashable.data(), hashable.size());
+						hash = { *static_cast<core::blake3_hash_t>(hasher).data };
+						lookupHash = std::bit_cast<uint64_t, uint8_t[8]>({hash[0], hash[1], hash[2], hash[3], hash[4], hash[5], hash[6], hash[7]});
+						for (auto i = 8u; i < 32; i++) {
 							core::hash_combine<uint64_t>(lookupHash, hash[i]);
```

Comment:
there's a `std::hash` specialization for `core::blake3_hash_t` that lets you make a STD-compatible size_T hash from the large hash